File: python/1_two_sum.py
from typing import List

# O(n^2)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[j] + nums[i] == target:
                    return [i,j]

                
s = Solution()
ans = s.twoSum([2,3,6,14,56], 9)
print(ans)

#O(n)
# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         num_dict = {}
#         for i, num in enumerate(nums):
#             if target - num in num_dict:
#                 return [num_dict[target - num], i]
#             num_dict[num] = i

File: cpp/1_two_sum.cpp
#include<iostream>
#include<vector>
using namespace std;


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        int n = nums.size();
        for(int i=0; i<n; i++){
            for(int j=i+1; j<n; j++){
                if (nums[i] + nums[j] == target) return {i,j};
            }
        }
        return {}; // add this line to return an empty vector if no solution is found
    }
};


int main(){
    Solution s;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = s.twoSum(nums, target);
    // print the result
    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    return 0;
}